accepted/2020/windows-specific-apis/windows-specific-apis.md

```diff
@@ -0,0 +1,475 @@
+# Marking Windows-only APIs
+
+As discussed in [dotnet/runtime#33331] we're adding an analyzer to detect usage
+of platform-specific APIs. This analyzer requires a custom attribute applied to
+method, containing type, or assembly.
+
+In this proposal I'm covering which APIs we mark as Windows-specific an how. It
+was largely informed by the work put into [platform-compat].
+
+[dotnet/runtime#33331]: https://github.com/dotnet/runtime/issues/33331
+[platform-compat]: https://github.com/dotnet/platform-compat
+
+## Which Windows version will we use?
+
+The lowest version of Windows that we support with .NET Core is Windows 7. Also,
+we generally don't expose functionality that requires a higher version of
+Windows.
+
+Thus, I propose that we use `7.0` as the minimum version number instead of
+hunting down which Windows version actually introduced a given feature, which
+means the custom attribute will look as follows:
+
+```C#
+[MinimumOSPlatform("windows7.0")]
+```
+
+## Entire assemblies
+
+There is a set of assemblies that we can mark wholesale as being
+Windows-specific:
+
+> **OPEN**: Should `System.Threading.Overlapped.dll` be removed?
+> **OPEN**: I was told that part of `DirectoryServices` is now cross-platform. Is that true?
+
+* **Microsoft.Win32.Registry.dll**
+* **Microsoft.Win32.Registry.AccessControl.dll**
+* **Microsoft.Win32.SystemEvents.dll**
+* **System.Diagnostics.EventLog.dll**
+* **System.Diagnostics.PerformanceCounter.dll**
+* **System.DirectoryServices.dll**
+* **System.DirectoryServices.AccountManagement.dll**
+* **System.DirectoryServices.Protocols.dll**
+* **System.IO.FileSystem.AccessControl.dll**
+* **System.IO.FileSystem.DriveInfo.dll**
+* **System.IO.Pipes.AccessControl.dll**
+* **System.Management.dll**
+* **System.Net.Http.WinHttpHandler.dll**
```

Comment:
It looks like we don't ship `net5.0` version of `System.Net.Http.WinHttpHandler.dll`  (missing `NetCoreAppCurrent` in targets):

https://github.com/dotnet/runtime/blob/a9944fa8a721af673999d074caf1dd74617f6ba6/src/libraries/System.Net.Http.WinHttpHandler/src/System.Net.Http.WinHttpHandler.csproj#L5

does it mean that this assembly should not be annotated with the new attribute?

Reply:
We still want to apply the attribute. Since we cannot reference the attribute from `net5.0` though, we will need to create an internal implementation of the attribute within this assembly using the same namespace and type name. This is the same approach that was taken for nullable annotations.